### PR TITLE
fix: rust bindings build for macos

### PR DESCRIPTION
**Pull request by knst**
That's some useful changes that blocked build, but it's not full final version yet.

Further included changes:

- Relic Arch fix (also in #90)
- Relic wasn't linked when building with feature "Apple"
- GMP was bumped to 6.3.0 (Since 6.2.1 have issues on m1: also dropped the custom patches for 6.2.1)
- Dropped `apple.rust.sh` and `apple.rust.single.sh` since only `apple.rust.deps.sh` is used.
- Fix for passing correctly CC and CXX paths when building for Apple

### CODE REVIEW

File: apple.rust.deps.sh [hunk 1]
```diff
@@ -143,7 +144,10 @@ CC="$CLANG" CFLAGS="$CFLAGS" CPPFLAGS="$CFLAGS" LDFLAGS="$CFLAGS" \
 --host=${HOST} --prefix="${CURRENT_DIR}/gmplib-${PFX}" \
 --disable-shared --enable-static --disable-assembly -v
 EOF
-
+    echo CC="$CLANG" CFLAGS="$CFLAGS" CPPFLAGS="$CFLAGS" LDFLAGS="$CFLAGS" \
```
Comment (kwvg) [inline apple.rust.deps.sh:147]: `sh "$CONFIGURESCRIPT"` should already run the `./configure` command, why do we need to run it again?

File: rust-bindings/bls-signatures/src/lib.rs [hunk 2]
```diff
@@ -101,4 +101,20 @@ mod tests {
...
+        let seed = b"seedweedseedweedseedweedseedweed";
+        let scheme = LegacySchemeMPL::new();
+
+        let private_key = PrivateKey::key_gen(&scheme, seed).expect("unable to generate private key");
+        let public_key = private_key.g1_element().expect("unable to get public key");
+
+        println!("private_key:{:?}", private_key.to_bytes().as_slice());
+        println!("public_key:{:?}", public_key.to_bytes().as_slice());
+
+        println!("ody done");
```
Comment (kwvg) [inline rust-bindings/bls-signatures/src/lib.rs:118]:
```suggestion
```

File: rust-bindings/bls-signatures/src/lib.rs [hunk 3]
```diff
@@ -101,4 +101,20 @@ mod tests {
...
+    fn test_bls_public_key() {
+        println!("ody");
+
+        let seed = b"seedweedseedweedseedweedseedweed";
+        let scheme = LegacySchemeMPL::new();
+
+        let private_key = PrivateKey::key_gen(&scheme, seed).expect("unable to generate private key");
+        let public_key = private_key.g1_element().expect("unable to get public key");
+
+        println!("private_key:{:?}", private_key.to_bytes().as_slice());
```
Comment (kwvg) [inline rust-bindings/bls-signatures/src/lib.rs:115]: Can we find the value of `sk`, hardcode it and test the generated value against it? So far the test only validates that a value is generated but is not referenced against a known-good value.

**Review by kwvg — COMMENTED** [review]
Consider cf62af3f9955eb475d4ae00b7636a277b1289180 for cleaned up history

File: rust-bindings/bls-signatures/src/lib.rs [hunk 4]
```diff
@@ -101,4 +101,20 @@ mod tests {
             grandchild_sk_u.g1_element().expect("cannot get public key")
         );
     }
+
+    #[test]
+    fn test_bls_public_key() {
+        println!("ody");
```
Comment (kwvg) [inline rust-bindings/bls-signatures/src/lib.rs:107]:
```suggestion
```

**Review by kwvg — COMMENTED** [review]
**EDIT:** Disregard this comment message

File: rust-bindings/bls-signatures/src/lib.rs (hunk 3 above)
Comment (UdjinM6) [reply rust-bindings/bls-signatures/src/lib.rs:115]: Agree. Should drop all println-s and instead compare the resulting privkey/pubkey to the expected one.

File: rust-bindings/bls-signatures/src/lib.rs (hunk 2 above)
Comment (knst) [reply rust-bindings/bls-signatures/src/lib.rs:118]: should you just completely remove this test? @ogabrielides

**Comment by HashEngineering** [comment]
here are my results - use https://github.com/dashpay/bls-signatures/pull/89 along with my own changes in https://github.com/dashpay/bls-signatures/pull/88 to build my system to support Android.  There are several layers:
* Rust running on Mac (ARM-64) - Yes, no issues
* Java running on Mac (ARM-64) - Yes, no issues
* Java running on Android (ARM-64) - Yes, no issues

**Review by ogabrielides — DISMISSED** [review]
ACK https://github.com/dashpay/bls-signatures/pull/89/commits/b4ca7789713cc5a1718a828433f0e0503bf71dfd

File: apple.rust.deps.sh [hunk 5]
```diff
@@ -143,7 +144,7 @@ CC="$CLANG" CFLAGS="$CFLAGS" CPPFLAGS="$CFLAGS" LDFLAGS="$CFLAGS" \
 --host=${HOST} --prefix="${CURRENT_DIR}/gmplib-${PFX}" \
 --disable-shared --enable-static --disable-assembly -v
 EOF
-
```
Comment (kwvg) [inline apple.rust.deps.sh:146]: nit: whitespace

**Review by kwvg — APPROVED** [review]
Please change the base branch to `develop`.

utACK b4ca7789713cc5a1718a828433f0e0503bf71dfd

**Review by ogabrielides — APPROVED** [review]
ACK https://github.com/dashpay/bls-signatures/commit/b4ca7789713cc5a1718a828433f0e0503bf71dfd

**Review by knst — COMMENTED** [review]
LGTM b4ca7789713cc5a1718a828433f0e0503bf71dfd

but I can't really self-review my own PR on github

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**